includes/fields.php

```diff
@@ -1065,19 +1068,24 @@ function pmpro_add_user_fields_to_email( $email ) {
 				}
 			}
 
-			// Remove any field groups that are not for the membership level that was purchased.
+			// Get the level ID from the email.
 			if ( ! empty( $email->data['membership_id'] ) ) {
 				$level_id = $email->data['membership_id'];
-				foreach ( $fields_groups as $key => $group ) {
-					if ( ! empty( $group->levels ) && ! in_array( $level_id, $group->levels ) ) {
-						unset( $fields_groups[ $key ] );
-					}
+			}
+			
+
+			// Remove any field groups that are not for the membership level that was purchased.
+			foreach ( $fields_groups as $key => $group ) {
+				if ( ! empty( $group->levels ) && ! in_array( $level_id, $group->levels ) ) {
+					unset( $fields_groups[ $key ] );
 				}
 			}
```

Comment:
Previously, we had the `foreach` inside the `if` in case `$email->data['membership_id'];`, and there for `$level_id`, is not set. Should we default `$level_id` to a value in this case to avoid PHP errors? We also check `$level_id` lower down in the `foreach` loop of fields, so we want to make sure to avoid errors in both of those cases